packages/ember-views/lib/system/event_dispatcher.js

```diff
@@ -290,27 +295,86 @@ export default EmberObject.extend({
+      // see https://github.com/jquery/jquery/blob/899c56f6ada26821e8af12d9f35fa039100e838e/src/event.js#L666-L700
+      if (EVENT_MAP[event] !== undefined) {
+        let mappedEventType = EVENT_MAP[event];
+        let origEventType = event;
+
+        let createFakeEvent = (eventType, event) => {
+          let fakeEvent = document.createEvent('MouseEvent');
+          fakeEvent.initMouseEvent(
+            eventType,
+            false,
+            false,
+            event.view,
+            event.detail,
+            event.screenX,
+            event.screenY,
+            event.clientX,
+            event.clientY,
+            event.ctrlKey,
+            event.altKey,
+            event.shiftKey,
+            event.metaKey,
+            event.button,
+            event.relatedTarget
+          );
+
+          // fake event.target as we don't dispatch the event
+          Object.defineProperty(fakeEvent, 'target', { value: event.target, enumerable: true });
+
+          return fakeEvent;
+        };
+
+        let handleMappedEvent = (this._eventHandlers[mappedEventType] = event => {
+          let target = event.target;
+          let related = event.relatedTarget;
 
-        do {
-          if (viewRegistry[target.id]) {
-            if (viewHandler(target, event) === false) {
-              event.preventDefault();
-              event.stopPropagation();
+          do {
+            // For mouseenter/leave call the handler if related is outside the target.
+            // No relatedTarget if the mouse left/entered the browser window
+            if (viewRegistry[target.id]) {
+              if (!related || (related !== target && !target.contains(related))) {
+                viewHandler(target, createFakeEvent(origEventType, event));
+              }
               break;
-            }
-          } else if (target.hasAttribute('data-ember-action')) {
-            if (actionHandler(target, event) === false) {
+            } else if (target.hasAttribute('data-ember-action')) {
+              if (!related || (related !== target && !target.contains(related))) {
+                actionHandler(target, createFakeEvent(origEventType, event));
+              }
               break;
             }
-          }
 
-          target = target.parentNode;
-        } while (target && target.nodeType === 1);
-      });
+            target = target.parentNode;
+          } while (target && target.nodeType === 1);
+        });
+
+        rootElement.addEventListener(mappedEventType, handleMappedEvent);
+      } else {
+        let handleEvent = (this._eventHandlers[event] = event => {
+          let target = event.target;
+
+          do {
+            if (viewRegistry[target.id]) {
+              if (viewHandler(target, event) === false) {
+                event.preventDefault();
+                event.stopPropagation();
+                break;
+              }
+            } else if (target.hasAttribute('data-ember-action')) {
```

Comment:
`target` can be a text node which does not support hasAttribute method -> http://take.ms/vCrm8C